### PR TITLE
Custom progress bar: PR 1

### PR DESCRIPTION
**Pull request by anisehsani**
### Overview
- Changed `ProgressBarLogger` to support `dataloader_label` 
- Added tests for individual progress bar loggers that track one `dataloader_label` each
- Progress bar loggers can track all data loaders if no `dataloader_label` is set, or just a specific dataloader's progress
- Progress bars can't be passed into the Trainer through `loggers` and also have `progress_bar` set to True

### Examples
Reusing Examples from the Progress Bar UX Refactor:

### Example 1

YAML:
```yaml
train_dataset:
  &train_dataset
  mnist:
    use_synthetic: true
evaluators:
  eval1:
    label: eval/batch
    eval_interval: 1ba
    subset_num_batches: 2
    eval_dataset:
      <<: *train_dataset
  eval2:
    label: eval/ep
    eval_interval: 1ep
    subset_num_batches: 2
    eval_dataset:
      <<: *train_dataset
model:
  mnist_classifier:
    num_classes: 10
max_duration: 2ep
train_batch_size: 2000
eval_batch_size: 2000
dataloader:
  pin_memory: true
  timeout: 0
  prefetch_factor: 2
  persistent_workers: true
  num_workers: 1
train_subset_num_batches: 2

progress_bar: False
loggers:
  - progress_bar:
      dataloader_label: eval/batch
      stream: stderr
  - progress_bar:
      dataloader_label: eval/ep
      stream: stderr
  - progress_bar:
      dataloader_label: train
      stream: stderr
```

```

eval/batch     Batch   1:  100%|█████████████████████████| 2/2 [00:00<00:00, 41.23ba/s]                                                                       
eval/batch     Batch   2:  100%|█████████████████████████| 2/2 [00:00<00:00, 45.96ba/s]                                                                       
train          Epoch   0:  100%|█████████████████████████| 2/2 [00:01<00:00,  1.43s/ba, loss/train=2.2941]                                                    
eval/ep        Epoch   0:  100%|█████████████████████████| 2/2 [00:00<00:00, 47.12ba/s]                                                                       
eval/batch     Batch   3:  100%|█████████████████████████| 2/2 [00:00<00:00, 45.63ba/s]                                                                       
eval/batch     Batch   4:  100%|█████████████████████████| 2/2 [00:00<00:00, 17.04ba/s]                                                                       
train          Epoch   1:  100%|█████████████████████████| 2/2 [00:00<00:00, 19.96ba/s, loss/train=2.2693]                                                    
eval/ep        Epoch   1:  100%|█████████████████████████| 2/2 [00:00<00:00, 49.97ba/s]     
```

Which is exactly the same output that is given when setting `progress_bar` to True and tracking everything. Changing the last part of the yaml to:
```yaml
progress_bar: False
loggers:
  - progress_bar:
      dataloader_label: eval/batch
      stream: stderr
```
Gives:
```

eval/batch     Batch   1:  100%|█████████████████████████| 2/2 [00:00<00:00, 45.36ba/s]                                                                       
eval/batch     Batch   2:  100%|█████████████████████████| 2/2 [00:00<00:00, 51.26ba/s]                                                                       
eval/batch     Batch   3:  100%|█████████████████████████| 2/2 [00:00<00:00, 46.27ba/s]                                                                       
eval/batch     Batch   4:  100%|█████████████████████████| 2/2 [00:00<00:00, 17.70ba/s]       
```

### Example 2

Have verified for each of the examples in PR #1264, replacing the default logger with individual progress bars for each metric simply restricts output to the progress bars specified via the `dataloader_label`, but is otherwise identical

### CODE REVIEW

File: composer/loggers/progress_bar_logger.py [hunk 1]
```diff
@@ -114,6 +114,9 @@ class ProgressBarLogger(LoggerDestination):
 
     Args:
         progress_bar (bool, optional): Whether to show a progress bar. (default: ``True``)
+        dataloader_label (str, optional): The label for the dataloader that the progress bar is tracking.
+
+            If no dataloader_label is specified, then the logger will track all dataloaders.
```
Comment (ravi-mosaicml) [inline composer/loggers/progress_bar_logger.py:119]:
```suggestion
            If no ``dataloader_label`` is specified, then the logger will track all dataloaders.
```

File: tests/loggers/test_progress_bar_logger.py [hunk 2]
```diff
@@ -66,6 +69,91 @@ def get_mock_tqdm(bar_format: str, *args: object, **kwargs: object):
...
+     Time.from_timestring('100sp'),
+     Time.from_timestring('5ba')],
+)
+@pytest.mark.timeout(10)
+def test_progress_bar_dataloader_label(max_duration: Time[int], monkeypatch: MonkeyPatch, world_size: int):
+
+    mock_tqdms_train = []
+    mock_tqdms_eval = []
+
+    def get_mock_tqdm(bar_format: str, *args: object, **kwargs: object):
```
Comment (ravi-mosaicml) [inline tests/loggers/test_progress_bar_logger.py:108]: Can this be a shared helper?

File: tests/loggers/test_progress_bar_logger.py [hunk 3]
```diff
@@ -66,6 +69,91 @@ def get_mock_tqdm(bar_format: str, *args: object, **kwargs: object):
 
     trainer.fit()
 
+    if dist.get_local_rank() != 0:
```
Comment (ravi-mosaicml) [inline tests/loggers/test_progress_bar_logger.py:72]: There seems to be some duplicative code between the two tests...curious would it be possible to refactor them to use shared helpers?

**Review by ravi-mosaicml — COMMENTED** [review]
Overall looks good! See comments. Would also like to do a bit more exploration to see if we can remove the class variable for the dummy pbar, but if not, let's document why it's important to create only one dummy pbar globally. Also curious if you have tested this (manually) over k8s?

**Comment by ravi-mosaicml** [comment]
Forgot to mention, but can the logic to print to the console (if `log_to_console=True`) also condition on the dataloader label? Otherwise the same statement could be printed multiple times.

**Review by ravi-mosaicml — COMMENTED** [review]
See comments; will take a closer look after the changes discussed offline.

**Comment by anisehsani** [comment]
#1303 is a replacement that implements this
